CONTRIBUTING.md

```diff
@@ -63,3 +65,15 @@ contributing to aks-engine. All issue types follow the same general lifecycle. D
 1. If you haven't already done so, sign a Contributor License Agreement (see details above).
 2. Fork the desired repo, develop and test your code changes.
 3. Submit a pull request.
+
+## Pull Requests and Generated Code
+
+**Summary:** Always run `make ensure-generated` before you submit a pull request. If that command
+returns an error, it will also have generated files that need to be included with the PR.
```

Comment:
Why would `ensure-generated` return an error if it also resulted in the needed generated files? That sounds like something that would return 0.

Reply:
Because if the command fails, the PR author has more to do (commit the files and try again). So it returns an error return code.

Since it had to generate the files in order to know if there's a diff, the simplest path is to have the user just commit the files already generated. Committing what could be seen as a side effect does seem odd, which was the motivation for renaming the make target to `ensure-generated`.